lib/resources/api_base_helper.dart

```diff
@@ -66,3 +67,59 @@ class ApiBaseHelper {
     }
   }
 }
+
+/*
+Interceptor that modify API Request by adding authentication information to the request.
+ */
+class AuthInterceptor implements InterceptorContract {
+
+  @override
+  Future<RequestData> interceptRequest({RequestData data}) async {
+    await addAuthHeader(data);
+    return data;
+  }
+
+  @override
+  Future<ResponseData> interceptResponse({ResponseData data}) async {
+    return data;
+  }
+
+  Future<void> addAuthHeader(RequestData data) async {
+    final String token = await AuthManager().getAccessToken();
+    if(token != null) {
+      data.headers['Authorization'] = token;
+    }
+  }
+}
+
+///
+/// Interceptor for debug purpose.
+///
+class LogInterceptor implements InterceptorContract {
```

Comment:
Can we move this class to its own separate file?

Reply:
Done the changes